projects/igniteui-angular/src/lib/date-range-picker/date-range-picker.component.ts

```diff
@@ -785,23 +846,23 @@ export class IgxDateRangePickerComponent extends DisplayDensityBase
             if (start && end) {
                 start.dateTimeEditor.valueChange
                     .pipe(takeUntil(this.$destroy))
-                    .subscribe((date: Date) => {
-                        this.updateValue({ start: date as Date, end: this.value?.end });
+                    .subscribe(() => {
+                        this.value = { start: start.dateTimeEditor.value, end: end.dateTimeEditor.value };
                     });
                 end.dateTimeEditor.valueChange
                     .pipe(takeUntil(this.$destroy))
-                    .subscribe((date: Date) => {
-                        this.updateValue({ start: this.value?.start, end: date as Date });
+                    .subscribe(() => {
+                        this.value = { start: start.dateTimeEditor.value, end: end.dateTimeEditor.value };
                     });
                 start.dateTimeEditor.validationFailed
                     .pipe(takeUntil(this.$destroy))
-                    .subscribe((event: IgxDateTimeEditorEventArgs) => {
-                        this.updateValue({ start: event.newValue as Date, end: this.value?.end });
+                    .subscribe(() => {
+                        this.value = { start: start.dateTimeEditor.value, end: end.dateTimeEditor.value };
```

Comment:
Why on earth do we update the range value on .. validationFailed?!

Reply:
I've cleaned this a little